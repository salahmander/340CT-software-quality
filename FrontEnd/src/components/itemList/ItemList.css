.ItemList {
    box-shadow: 0px 5px 30px lime;
    background-color: rgba(255, 255, 255, 0.9);
    color: black;
    width: 70%;
    transition: 0.3s;
    border-radius: 5px;
    display: flex;
    margin-bottom: 5%;
    margin-left: 15%;
   
}

.ItemList:hover {
    box-shadow: 0px 10px 50px lime;
}

.itemPrice{
   margin-top:145px;
   font-size: 200%;
   list-style-type: none;
}

img {
    border-radius: 5px 5px 0 0;
    width: 60%;
}

.container {
    padding: 2px 16px;
}
.linkButton {
    background:none!important;
    color:inherit;
    border:none; 
    padding:0!important;
    font: inherit;
    /*border is optional*/
    border-bottom:0 solid #444; 
    border-left: 05px solid #83B81A;
    padding: 10px 10px;
    border-radius: 5px;
    outline: none;
    cursor: pointer;
}






