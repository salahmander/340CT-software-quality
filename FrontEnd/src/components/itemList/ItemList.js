import React, { Component } from 'react';
import './ItemList.css';


class ItemList extends Component {
    constructor(props) {
        super(props);

        this.onClickHandler = this.onClickHandler.bind(this);
    }

    onClickHandler(event) {
        event.preventDefault();
        this.props.onClick(this.props.id);
        console.log(this.props.id)
    }


    render() {

        return (
            <div className="ItemList">
                <img src={this.props.image} alt={this.props.imgAlt} />
                <div className="container">
                    <button onClick={this.onClickHandler} className="linkButton"><h4><b>{this.props.title}</b></h4></button>
                    <p>{this.props.description}</p>
                    
                    <ul>
                        <li>CPU: {this.props.CPU} </li>
                        <li>RAM: {this.props.RAM} </li>
                        <li>Graphics card: {this.props.GPU}</li>
                        <li>Storage: {this.props.Storage}</li>
                        <li>Storage size: {this.props.Size}</li>
                    </ul>

                    <span className ="itemPrice">Price: Â£{this.props.price}</span>
                </div>
            </div>
        );
    }
}
export default ItemList;