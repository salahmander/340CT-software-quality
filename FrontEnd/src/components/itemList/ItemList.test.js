import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';

import ItemList from './ItemList'

describe('item list ', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ItemList />, div);
        ReactDOM.unmountComponentAtNode(div);
      });

    it('should render correctly in "debug" mode and match the snapshot', () => {
        const component = shallow(<ItemList debug />);
        expect(component).toMatchSnapshot();
      });
      
    });