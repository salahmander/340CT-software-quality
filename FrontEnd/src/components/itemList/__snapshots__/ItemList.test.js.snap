// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`item list  should render correctly in "debug" mode and match the snapshot 1`] = `
<div
  className="ItemList"
>
  <img />
  <div
    className="container"
  >
    <button
      className="linkButton"
      onClick={[Function]}
    >
      <h4>
        <b />
      </h4>
    </button>
    <p />
    <ul>
      <li>
        CPU: 
         
      </li>
      <li>
        RAM: 
         
      </li>
      <li>
        Graphics card: 
      </li>
      <li>
        Storage: 
      </li>
      <li>
        Storage size: 
      </li>
    </ul>
    <span
      className="itemPrice"
    >
      Price: Â£
    </span>
  </div>
</div>
`;
